doc/contribute/devtheory.rst

```diff
@@ -0,0 +1,185 @@
+--------
+
+Every geometry contains entities that are either volumes, surfaces, and curves.
+There are two types of relationships that can relate entities to other entities.
+The first is called a parent-child relationship. Volumes are parents to surfaces
+that make up that volume; surfaces are parents of of curves; and curves are
+parents to the geometric vertices of that curve.
+
+The second type of relationship is the set relationship, which is different
+from a parent-child relationship. Each surface and curve has an entity set.
+The entity set for a surface contains the triangles and their vertices for that
+surface. Curves contain edges and their vertices. The entity set for a volume,
+however, is empty. While a volume is made of surfaces (the parent-child
+relationship), the volume does not contain an entity set.
+
+Sense Tags
+----------
+
+Each surface is tagged with the two volume handles of the adjacent volumes.
+The first of the two surfaces is designated as the forward direction and the
+second is designated with the reverse direction. It is important to note that
+these surfaces senses may not be consistent with how an MC code determines
+the surface sense.
+
+Indices
+~~~~~~~
+
+Types of Indices
+----------------
+
+There are three ways to identify any surface and volume in a geometry:
+global ID, index, or entity handle. The global ID is an integer number that
+corresponds to the Cubit ID. As a result, the set of global IDs does not
+necessarily have to be contiguous. During setup, the global ID is tagged on the
+entity set. The second method is to identify by index. This is an ordinal
+numerical reference for each volume or surface in the geometry. The third method
+is to identify by the entity handle of the entity set. The entity handle is
+assigned by MOAB and is not necessarily contiguous.
+
+Cross Referencing
+-----------------
+
+If one type of indice is known, then either of the other two can be determined.
+By using the functions below or a combination of the functions, either ID,
+index, or entity handle can be determined. It is important to note that surfaces
+and volumes each have their own set of integer indexes and global IDs, while
+entity handles are unique to each entity set. Each integer (whether it is an
+index or global ID) can refer to multiple entity handles (a surface or volume).
+Therefore, when cross referencing by either index or global ID, the dimension
+needs to be supplied, but is not necessary when cross referencing by entity
+handle. There are five methods for cross referencing: entity handle by index
+(entity_by_index), global ID by index (id_by_index), entity handle by global
+ID (entity_by_id), global ID by entity handle (get_entity_id), index by entity
+handle (index_by_handle).
+
+OBB Tree
+~~~~~~~~
+
+What is an OBB?
+---------------
+
+An OBB, or Oriented Bounding Box, is a box that contains a single facet or set
+of facets and has an axis oriented such that it is the smallest possible box
```

Comment:
I think I correctly clarified what you are talking about now, but can you review the new statement and let me know if it is or is not correct now.